bindings/engine.i

```diff
@@ -160,6 +166,50 @@ SplitsVec gnc_get_match_commodity_splits (AccountVec accounts, bool use_end_date
     return rv;
 }
 
+static SCM
+accumulate_splits_by_dates (const SplitsVec& splits, SCM dates, SCM last_result,
+                            std::function<bool(Split*,SCM)> no_later_than_date,
+                            std::function<SCM(Split*)> get_result)
+{
+    SCM rv = SCM_EOL;
+    for (auto splits_it = splits.begin(); !scm_is_null(dates); dates = scm_cdr (dates))
```

Comment:
That's really obtuse. One expects a for loop to iterate on the variable in the first clause. 
```
auto splits_it{splits.begin()};
for (; !scm_is_null(dates); dates = scm_cdr (dates))
{
```
is more idiomatic.